multimodal/src/autogluon/multimodal/data/preprocess_dataframe.py

```diff
@@ -265,7 +293,6 @@ def transform_text(
         -------
         All the text data stored in a dictionary.
         """
-        assert self._fit_called, "You will need to first call preprocessor.fit before calling preprocessor.transform."
```

Comment:
Should we remove the assertion here? We can still call `.fit()` even if it's the zero-shot scenario. We may just ensure that `y=None` when calling .fit(). For example, `.fit(X, Y=None)`.

Reply:
There is another situation (maybe out of scope), that is doing self-supervised learning. Take contrastive learning as an example, we only have `X` as input, both positive and negative samples are from the same `X`. So there is no `Y`

Reply:
Calling `fit()` has no effect on image and text data. It is mainly for the tabular data. That's why I remove the assertions in the `transform_image()` and `transform_text()`.

Reply:
Yeah, later on, we may consider to do something for Image/Text. Removing the assertion is fine for now (since our focus is to add more functionality) but we may consider to refactor the code later.

Reply:
Refactored `fit()` into `fit_x()` and `fix_y()` and added back the assertions.